core/src/main/java/org/testcontainers/containers/output/WaitingConsumer.java

```diff
@@ -113,8 +113,12 @@ private void waitUntil(Predicate<OutputFrame> predicate, long expiry, int times)
     /**
      * Wait until Docker closes the stream of output.
      */
-    public void waitUntilEnd() throws TimeoutException {
-        waitUntilEnd(Long.MAX_VALUE);
+    public void waitUntilEnd() {
+        try {
+            waitUntilEnd(Long.MAX_VALUE);
+        } catch (TimeoutException ignored) {
+            // timeout condition can never occur in a realistic timeframe
```

Comment:
`@SneakyThrows`?

Reply:
Hmm... For people who don't know Lombok, `@SneakyThrows` is a bit more surface area to know about, and they'd have to know about it now.

Swallowing the exception would be a little surprising, but I'd imagine the universe ending would be a bigger surprise in the mean time 😉.

I'll change it to re-throw wrapped in an `IllegalStateException`, given that the timeout is effectively an impossible situation to find ourselves in 😄